Paintroid/src/main/java/org/catrobat/paintroid/listener/DrawingSurfaceListener.kt

```diff
@@ -93,11 +93,11 @@ open class DrawingSurfaceListener(
         autoScrollTask.setViewDimensions(view.width, view.height)
     }
 
-    private fun handleActionMove(currentTool: Tool, view: View, event: MotionEvent) {
+    private fun handleActionMove(currentTool: Tool?, view: View, event: MotionEvent) {
         val xOld: Float
         val yOld: Float
         if (event.pointerCount == 1) {
-            if (currentTool.handToolMode()) {
+            if (currentTool != null && currentTool.handToolMode()) {
```

Comment:
If the tool is null, should this function even be executed? 🤔

Reply:
If you are talking about the handleActionMove method, I think this should be executed, but I tried to improve it a bit. Please take a look at it again :)